packages/material/src/controls/MaterialSimpleAnyOfControl.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+  The MIT License
+
+  Copyright (c) 2018-2019 EclipseSource Munich
+  https://github.com/eclipsesource/jsonforms
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import {
+  and,
+  ControlProps,
+  ControlState,
+  EnumCellProps,
+  JsonSchema,
+  mapDispatchToControlProps,
+  mapStateToControlProps,
+  RankedTester,
+  rankWith,
+  schemaMatches,
+  uiTypeIs,
+  WithClassname
+} from '@jsonforms/core';
+import { Control } from '@jsonforms/react';
+import { Input } from '@material-ui/core';
+import { InputBaseComponentProps } from '@material-ui/core/InputBase';
+import merge from 'lodash/merge';
+import React from 'react';
+import { connect } from 'react-redux';
+import { MaterialInputControl } from './MaterialInputControl';
+
+const MuiAutocompleteInputText = (props: EnumCellProps & WithClassname) => {
+  const {
+    data,
+    config,
+    className,
+    id,
+    enabled,
+    uischema,
+    isValid,
+    path,
+    handleChange,
+    schema
+  } = props;
+
+  const findEnumSchema = (schemas: JsonSchema[]) =>
```

Comment:
Hoist `findEnumSchema` and `findTextSchema` out of render component, re-use in tester.